public/calendar_reboot.js

```diff
@@ -828,21 +933,35 @@ var LocalCalendarStorage = function(params) {
         var randomNumber = Math.random() * 4000;
         setTimeout(runFunc, randomNumber);
         
+
     };
     
     self.getAllCalendarIds = function() {
         // Returns a promise for the allCalendarIds object from storage
         
         return new Promise( function(resolve, reject) {
+            self.startWork();
             var allCalendarIds = loadFromLocalStorage(toKey(allCalendarIdsKey));
             
             if (allCalendarIds !== null ) {
+
                 jitter(resolve, allCalendarIds);
+
             }
             else {
                 jitter(reject, "Not found");
             }
         })
+        .then( function(ids) {
+            console.log("getAllCalendarIds then function running");
+            self.endWork(); // should I actually be ending work here?
```

Comment:
Yes. The purpose of these is to indicate whether or not work is *still happening* and will finish sometime in the future. If you call `startWork` without calling `endWork` you will appear to be stuck in the "activity" state forever. You need to make sure that no matter what happens, `startWork` and `endWork` get called exactly the same number of times.

Also, don't forget to remove this comment.